src/openvic-simulation/economy/FactoryProducer.hpp

```diff
@@ -0,0 +1,49 @@
+#pragma once
+
+#include <cstdint>
+
+#include "openvic-simulation/economy/Good.hpp"
+#include "openvic-simulation/economy/ProductionType.hpp"
+#include "openvic-simulation/types/fixed_point/FixedPoint.hpp"
+#include "openvic-simulation/utility/Getters.hpp"
+
+namespace OpenVic {
+	class FactoryProducer final {
+	private:
+		static constexpr uint8_t DAYS_OF_HISTORY = 7;
+		uint8_t PROPERTY(profit_history_current);
+		std::array<fixed_point_t, DAYS_OF_HISTORY> PROPERTY(daily_profit_history);
+		ProductionType const& PROPERTY(production_type);
+		fixed_point_t PROPERTY(revenue_yesterday);
+		fixed_point_t PROPERTY(output_quantity_yesterday);
+		fixed_point_t PROPERTY(unsold_quantity_yesterday);
+		fixed_point_t PROPERTY(size_multiplier);
+		ordered_map<Pop*, Pop::pop_size_t> PROPERTY(employees);
+		Good::good_map_t PROPERTY(stockpile);
+		fixed_point_t PROPERTY(budget);
+		fixed_point_t PROPERTY(balance_yesterday);
+		fixed_point_t PROPERTY(received_investments_yesterday);
+		fixed_point_t PROPERTY(market_spendings_yesterday);
+		fixed_point_t PROPERTY(paychecks_yesterday);
```

Comment:
I would suggest using `wages` instead of `paychecks`, to match Vic2's terminology.

Reply:
Both the UI and savegame format use paychecks. See http://www.amicabile.com/v2/f1/B2.jpg
I'll leave it paychecks for now.